stdlib/Distributed/src/remotecall.jl

```diff
@@ -247,11 +247,14 @@ function del_clients(pairs::Vector)
     end
 end
 
-const any_gc_flag = Condition()
+const any_gc_flag = Threads.Condition()
 function start_gc_msgs_task()
-    @async while true
-        wait(any_gc_flag)
-        flush_gc_msgs()
+    @async begin
+        lock(any_gc_flag)
```

Comment:
To be idiomatic we should still have a try around the while loop with a finaliy that unlocks the lock.